packages/react-components/react-dialog/src/components/Dialog/renderDialog.tsx

```diff
@@ -14,7 +16,16 @@ export const renderDialog_unstable = (state: DialogState, contextValues: DialogC
     <DialogProvider value={contextValues.dialog}>
       <DialogSurfaceProvider value={contextValues.dialogSurface}>
         {trigger}
-        {content}
+        <Transition
```

Comment:
Not blocking, just pointing out another place to put the Transition. 

You could render the Transition in DialogSurface, and pass the open state in context instead of passing the transition status.

Reply:
if I do that I would not have access to the transition status on the surface style hook.

Reply:
well you could store the styles in state and apply them in the render function with a pure function

Reply:
I don't think it makes a huge difference, it's why I said it's up to you. :)